### PR TITLE
fix(sql): average function overflow for LONG column type

### PR DESCRIPTION
**Pull request by eugenels**
This PR adds  __int128 backed implementations for sum/avg methods to Rosti.
For now they are used by AvgLongVectorAggregateFunction only.

Fixes https://github.com/questdb/questdb/issues/1803

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|4|5|80.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|4|5|80.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 8 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|5|8|62.50%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|7|8|87.50%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 8 (62.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|5|8|62.50%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|7|8|87.50%|

File: core/src/test/java/io/questdb/griffin/engine/functions/groupby/AvgLongVecGroupByFunctionFactoryTest.java [hunk 1]
```diff
@@ -87,4 +87,29 @@ public void testSimple() throws Exception {
...
+            assertSql("select sym, avg(x) from test where x > 0 order by sym", expected);
+
+            final String diffExpected = "sym\tcolumn\n" +
+                    "a\ttrue\n" +
+                    "b\ttrue\n" +
+                    "c\ttrue\n";
+
+            // Here 6000 is a hack.
+            // Difference between avg(double) and avg(long) depends on column values range and rows count.
+            assertSql("with a as (select sym, avg(x) from test), b as (select sym, avg(x) from test where x > 0) " +
```
[the remark this answers is not in the document]
Comment (eugenels) [reply core/src/test/java/io/questdb/griffin/engine/functions/groupby/AvgLongVecGroupByFunctionFactoryTest.java:111]: Will fix that with a separate PR: https://github.com/questdb/questdb/issues/1832

**Review by ideoma — COMMENTED** [review]
Do you mind fixing `AvgIntVectorAggregateFunction` too?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|7|8|87.50%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|7|8|87.50%|

**Comment by eugenels** [comment]
> Do you mind fixing AvgIntVectorAggregateFunction too?
AvgInt implementation accumulates sum into long variable.

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 8 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|7|8|87.50%|

**Pull request merged**